src/client/pages/Offer/Introduction/Sidebar/CampaignCodeModal.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react'
```

Comment:
This is a copy of `OfferNew/Sidebar/DiscountCodeModal.tsx` with some modifications:
- renaming `OfferNew/Sidebar/DiscountCodeModal.tsx` --> `Offer/Sidebar/CampaignCodeModal.tsx`
- changed the order of the imports to a way I think it's more organized
- changed `internface Props` --> `export type CampaignCodeModalProps`
- changed component API:

    _Old_: 
    ```typescript
        interface Props {
            isOpen: boolean
            close: () => void
            refetch: () => Promise<void>
        }
    ```
    _New_:
    ```typescript
         export type CampaignCodeModalProps = {
             isOpen: boolean
             close: () => void
             onAddCampaignCode: (code: string) => Promise<{ hasError: boolean }>
        }
    ```
- removed `CookieStorage` related code
- added some logic to `resetForm`